libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp

```diff
@@ -317,26 +332,29 @@ bool AP_RangeFinder_LightWareI2C::sf20_init()
         return false;
     }
 
-    // Switches distance streaming on or off:
-    // 0 = off
-    // 1 = on
-    if (!sf20_send_and_expect("#SU,1\r\n", "su:1")) {
-        return false;
-    }
-
     // Changes the laser state:
     //    0 = laser is off
     //    1 = laser is running
     if (!sf20_send_and_expect("#LF,1\r\n", "lf:1")) {
         return false;
     }
 
+    // Switches distance streaming on or off:
+    // 0 = off
+    // 1 = on
+    if (!sf20_send_and_expect("#SU,1\r\n", "su:1")) {
+        return false;
+    }
+
     // Requests the measurement specified in the first stream.
     write_bytes((uint8_t*)init_stream_id[0], strlen(init_stream_id[0]));
 
     // call timer() at 20Hz
-    _dev->register_periodic_callback(50000,
-                                     FUNCTOR_BIND_MEMBER(&AP_RangeFinder_LightWareI2C::sf20_timer, void));
+    // Create this periodic calback once only
+    if (!sf20_periodic_callback_handle_) {
```

Comment:
Can the check for callback handle be a boolean or is this handle private member used elsewhere? I couldn't find a use

Reply:
This can be a static boolean. Will fix.